File: Interpreter C#/Interpreter/MyDictionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class MyDictionary<T1, T2> : MyIDictionary<T1, T2>
        {
            Dictionary<T1, T2> dict;
            public MyDictionary()
            {
                this.dict= new Dictionary<T1, T2>();
            }


            void MyIDictionary<T1, T2>.add(T1 key, T2 value)
            {
                this.dict.Add(key, value);
            }

            

            int MyIDictionary<T1, T2>.containsKey(T1 key)
            {
                if(this.dict.ContainsKey(key)==true)
                { return 1;}
                else
                { return 0; }
            }

            int MyIDictionary<T1, T2>.containsValue(T2 value)
            {
                if (this.dict.ContainsValue(value) == true)
                { return 1; }
                else
                { return 0; }
            }

            T2 MyIDictionary<T1, T2>.lookup(T1 key)
            {
                T2 val;
                this.dict.TryGetValue(key, out val);
                return val;
            }

            void MyIDictionary<T1, T2>.remove(T1 key)
            {
                this.dict.Remove(key);
            }

            
            void MyIDictionary<T1,T2>.update(T1 key, T2 value)
            {
                this.dict[key] = value;
            }
            public override string ToString()
            {
                string str="";
                foreach(KeyValuePair < T1, T2 > entry in this.dict)
                {
                    string s1 = entry.Key.ToString();
                    string s2 = entry.Value.ToString();
                    str =str+s1 + "=" + s2+"\n";        
                   
                }
                return str;
            }

            
        }
    }
}

File: MAP/Interpretor2/src/Model/MyDictionary.java
package Model;

import java.util.Enumeration;
import java.util.Hashtable;
import java.util.Map;
import java.util.Map.Entry;
import java.util.Set;


public abstract class MyDictionary<T1,T2> implements MyIDictionary<T1, T2> {
	Map<T1,T2> dict;
	
public MyDictionary(){
	this.dict=new Hashtable<T1,T2>();
}

public Object add(T1 key, T2 value)
{
	return this.dict.put(key,value);
}
public Object remove(T1 key)
{
	return this.dict.remove(key);
}
public int size()
{
	return this.dict.size();
}
public boolean containsKey(T1 key)
{
	return this.dict.containsKey(key);
}
public boolean containsValue(T2 value)
{
	return this.dict.containsValue(value);
}
public T2 lookup(T1 id)
{
	return (T2)dict.get(id);
}
public void update(T1 key,T2 value)
{
	this.dict.replace(key, value);
}
@Override
public void setContent(Map<T1,T2> m)
{
	this.dict=m;
	
}
@Override
public Map<T1,T2> getContent()
{
return this.dict;	
}

public Map<T1,T2> getMap()
{
	return this.dict;
}
public Set<T1> getKeySet()
{
	return this.dict.keySet();
}

public Set<Entry<T1, T2>> entrySet()
{
	return this.dict.entrySet();
}

}
